Assets/Scripts/SS3D/Systems/Roles/RoleSystem.cs

```diff
@@ -0,0 +1,195 @@
+        {
+            RoleCounter assistantRole = roleCounters.FirstOrDefault(rc => rc.role.Name == "Assistant");
+            RoleCounter securityRole = roleCounters.FirstOrDefault(rc => rc.role.Name == "Security");
+
+            if (securityRole == null || securityRole.currentRoles == securityRole.availableRoles)
+            {
+                assistantRole.AddPlayer(soul);
+                rolePlayers.Add(soul, assistantRole.role);
+            }
+            else
+            {
+                securityRole.AddPlayer(soul);
+                rolePlayers.Add(soul, securityRole.role);
+            }
+        }
+
+        /// <summary>
+        /// Remove players from the Role Counters if he quit before embarking
+        /// </summary>
+        /// <param name="soul"></param>
+        private void RemovePlayerFromCounters(Soul soul)
+        {
+            KeyValuePair<Soul, RoleData>? rolePlayer =
+                rolePlayers.FirstOrDefault(rp => rp.Key == soul);
+
+            if (rolePlayer != null)
+            {
+                var roleData = rolePlayer.Value;
+                var roleCounter = roleCounters.First(rc => rc.role == roleData.Value);
+
+                roleCounter.RemovePlayer(soul);
+            }
+        }
+
+        /// <summary>
+        /// Checks the role of the player and spawns his items
+        /// </summary>
+        /// <param name="entity">The player that will receive the items</param>
+        private void GiveRoleLoadoutToPlayer(Entity entity)
+        {
+            KeyValuePair<Soul, RoleData>? rolePlayer =
+                rolePlayers.FirstOrDefault(rp => rp.Key == entity.Mind.Soul);
+
+            if (rolePlayer != null)
+            {
+                var roleData = rolePlayer.Value;
+
+                Punpun.Say(this, entity.Ckey + " embarked with role " + roleData.Value.Name);
+                if (roleData.Value.Loadout != null)
+                {
+                    SpawnLoadoutItems(entity, roleData.Value.Loadout);
+                }
+            }
+        }
+
+        /// <summary>
+        /// Spawn all the role items for the player
+        /// </summary>
+        /// <param name="entity">The player that will receive the items</param>
+        /// <param name="loadout">The loadout of items he will receive</param>
+        private void SpawnLoadoutItems(Entity entity, RoleLoadout loadout)
+        {
+            var hands = entity.GetComponent<Hands>();
+            var inventory = entity.GetComponent<Inventory.Containers.Inventory>();
+
+            SpawnItemInSlot(loadout.leftHandItem, loadout.leftHand, hands.HandContainers[0].Container);
+            SpawnItemInSlot(loadout.rightHandItem, loadout.rightHand, hands.HandContainers[1].Container);
+            SpawnItemInSlot(loadout.leftPocketItem, loadout.leftPocket, inventory.LeftPocketContainer);
+            SpawnItemInSlot(loadout.rightPocketItem, loadout.rightPocket, inventory.RightPocketContainer);
+        }
```

Comment:
This is very hardcoded, can't we improve it?

Reply:
Possibly, but i'm not sure if it's needed, the player's inventory will always have pretty much the same amount of slots, it would only be a different case if a player started the round with many arms (the system only runs at round start), in which case it could just be modified to support many hands